Server/mods/deathmatch/logic/lua/CLuaManager.cpp

```diff
@@ -155,91 +155,185 @@ CResource* CLuaManager::GetVirtualMachineResource(lua_State* luaVM)
-    CLuaCFunctions::AddFunction("getWeaponOwner", CLuaFunctionDefs::GetWeaponOwner);
-    CLuaCFunctions::AddFunction("setWeaponFlags", CLuaFunctionDefs::SetWeaponFlags);
-    CLuaCFunctions::AddFunction("getWeaponFlags", CLuaFunctionDefs::GetWeaponFlags);
-    CLuaCFunctions::AddFunction("setWeaponFiringRate", CLuaFunctionDefs::SetWeaponFiringRate);
-    CLuaCFunctions::AddFunction("getWeaponFiringRate", CLuaFunctionDefs::GetWeaponFiringRate);
-    CLuaCFunctions::AddFunction("resetWeaponFiringRate", CLuaFunctionDefs::ResetWeaponFiringRate);
-    CLuaCFunctions::AddFunction("getWeaponAmmo", CLuaFunctionDefs::GetWeaponAmmo);
-    CLuaCFunctions::AddFunction("getWeaponClipAmmo", CLuaFunctionDefs::GetWeaponClipAmmo);
-    CLuaCFunctions::AddFunction("setWeaponClipAmmo", CLuaFunctionDefs::SetWeaponClipAmmo);
-#endif
-
-    // Console funcs
-    CLuaCFunctions::AddFunction("addCommandHandler", CLuaFunctionDefs::AddCommandHandler);
-    CLuaCFunctions::AddFunction("removeCommandHandler", CLuaFunctionDefs::RemoveCommandHandler);
-    CLuaCFunctions::AddFunction("executeCommandHandler", CLuaFunctionDefs::ExecuteCommandHandler);
-    CLuaCFunctions::AddFunction("getCommandHandlers", CLuaFunctionDefs::GetCommandHandlers);
-
-    // Server standard funcs
-    CLuaCFunctions::AddFunction("getMaxPlayers", CLuaFunctionDefs::GetMaxPlayers);
-    CLuaCFunctions::AddFunction("setMaxPlayers", CLuaFunctionDefs::SetMaxPlayers);
-    CLuaCFunctions::AddFunction("outputChatBox", CLuaFunctionDefs::OutputChatBox);
-    CLuaCFunctions::AddFunction("outputConsole", CLuaFunctionDefs::OutputConsole);
-    CLuaCFunctions::AddFunction("outputDebugString", CLuaFunctionDefs::OutputDebugString);
-    CLuaCFunctions::AddFunction("outputServerLog", CLuaFunctionDefs::OutputServerLog);
-    CLuaCFunctions::AddFunction("getServerName", CLuaFunctionDefs::GetServerName);
-    CLuaCFunctions::AddFunction("getServerHttpPort", CLuaFunctionDefs::GetServerHttpPort);
-    CLuaCFunctions::AddFunction("getServerPassword", CLuaFunctionDefs::GetServerPassword);
-    CLuaCFunctions::AddFunction("setServerPassword", CLuaFunctionDefs::SetServerPassword);
-    CLuaCFunctions::AddFunction("getServerConfigSetting", CLuaFunctionDefs::GetServerConfigSetting);
-    CLuaCFunctions::AddFunction("setServerConfigSetting", CLuaFunctionDefs::SetServerConfigSetting, true);
-    CLuaCFunctions::AddFunction("clearChatBox", CLuaFunctionDefs::ClearChatBox);
-
-    CLuaCFunctions::AddFunction("shutdown", CLuaFunctionDefs::shutdown, true);
-
-    // Loaded map funcs
-    CLuaCFunctions::AddFunction("getRootElement", CLuaFunctionDefs::GetRootElement);
-    CLuaCFunctions::AddFunction("loadMapData", CLuaFunctionDefs::LoadMapData);
-    CLuaCFunctions::AddFunction("saveMapData", CLuaFunctionDefs::SaveMapData);
+    std::map<const char*, lua_CFunction> functions{
+        {"addEvent", CLuaFunctionDefs::AddEvent}, {"addEventHandler", CLuaFunctionDefs::AddEventHandler},

```

Comment:
It's a good idea to have it on one line, but unfortunately, these lines are too long and it turns out it can be a bit awkward to read. Please move to individual lines!